rslib/src/storage/sqlite.rs

```diff
@@ -190,6 +192,31 @@ fn add_regexp_tags_function(db: &Connection) -> rusqlite::Result<()> {
     )
 }
 
+/// Adds sql function exclude_fields(flds, indices...)
+/// to return `flds` with fields whose indices are in `indices` removed
+fn add_exclude_fields_function(db: &Connection) -> rusqlite::Result<()> {
+    db.create_scalar_function(
+        "exclude_fields",
+        -1,
+        FunctionFlags::SQLITE_DETERMINISTIC,
+        move |ctx| {
+            assert!(ctx.len() > 1, "not enough arguments");
+
+            let joined_fields = ctx.get_raw(0).as_str()?;
+            let fields: Vec<&str> = joined_fields.split('\x1f').collect();
```

Comment:
The .collect() creates a vector that you don't need - you could remove this line and replace 'fields.iter()' below with joined_fields.split('\x1f'). This may improve performance somewhat.